### ISSUE TITLE
Allow onExecuteDone to be async

### ISSUE THREAD

**Issue by EmrysMyrddin** [dotansimha/envelop#868]
**Is your feature request related to a problem? Please describe.**

I would like to do some async work in the `onExecuteDone` hook. This is not possible for now because it can't return a promise.

**Describe the solution you'd like**

Looking to the code, I see no obvious problem to simply await the call of the onExecuteDone callback, but perhaps I'm missing some background or tricky implication of this.

**Describe alternatives you've considered**

I don't see any other way to do it.

**Additional context**

Here an example of use with a plugin which automatically wraps any query or mutation in a database transaction.

```ts
export function useTransactions(options): TransactionsPlugin {
  return {
    async onExecute({ args: { contextValue } }) {
      const operationType = getOperationType(contextValue)
      if(operationType != 'query' && operationType != 'mutation') return
      
	  const transaction = await options.newTransaction(operationType === 'query' ? 'read-only' : 'auto')
      
      return {
        async onExecuteDone({result}) {
          try {
          	if(isAsyncIterable(result)) throw TypeError('Async Iterable results are not implemented')
            if(result.errors) throw result.errors
            await transaction.commit()
          } catch (err) {
            await transaction.rollback(err)
          } finally {
            await transaction.releaseClient()
          }
        }
      }
    }
  }
}

```

**Comment by EmrysMyrddin** [comment]
I have made a PR with my solution, it seems that all tests are passing, which confort me in the idea that adding this shouln't break anything.

**Comment by dotansimha** [comment]
Thanks @EmrysMyrddin ! Merged and I'll cut a release soon :)

**Comment by dotansimha** [comment]
Available in `@envelop/core@1.4.0`.

**Comment by nlepage** [comment]
Thanks a lot @dotansimha for your reactivity.
